src/main/java/core/basesyntax/service/impl/FruitListImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.FruitTransaction;
+import core.basesyntax.service.FruitList;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FruitListImpl implements FruitList {
+    @Override
+    public List<FruitTransaction> getFruitList(List<FruitTransaction> list) {
+        List<String> listOfFruitsString = list.stream()
+                .map(FruitTransaction::getFruit)
+                .distinct()
+                .toList();
+        List<FruitTransaction> listOfFruitsAsObjects = new ArrayList<>();
+        for (String fruit : listOfFruitsString) {
+            int defaultQuantity = 0;
+            listOfFruitsAsObjects.add(new FruitTransaction(fruit, defaultQuantity));
+        }
+        return listOfFruitsAsObjects;
```

Comment:
The `getFruitList` method could be simplified. Instead of creating a list of strings and then converting them back into `FruitTransaction` objects, consider directly mapping to `FruitTransaction` with the distinct fruit names. Also, the default quantity could be set as a constant to avoid magic numbers.